src/validator/data_model.py

```
@@ -92,8 +92,8 @@ def set_idf(cls, idd_file: Path | str, idf_path: Path | None = None) -> None:
     @classmethod
     def _process_idf_field(cls) -> IDDField:
         if cls._idf is None:
-            raise ValueError(
-                "IDF is not set. Please set the IDF using BaseSchema.set_idf."
+            raise RuntimeError(
+                "IDF has not been initialized. Call IDFBaseModel.initialize() before using _process_idf_field."
             )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)